src/core/evaluator.js

```diff
@@ -1136,13 +1136,11 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
               break;
             case OPS.setFillCMYKColor:
               stateManager.state.fillColorSpace = ColorSpace.singletons.cmyk;
-              args = ColorSpace.singletons.cmyk.getRgb(args, 0);
-              fn = OPS.setFillRGBColor;
+              args.push(...ColorSpace.singletons.cmyk.getRgb(args, 0));
```

Comment:
One problem with these changes is that you're now passing completely unparsed/non-validated data to the main-thread, which you do *not* want to do since it will throw Errors on `postMessage` if the `args` isn't of the expected format (which could happen for e.g. corrupt documents).

Trying to address this would (again) add runtime overhead, despite it being completely unused in the default PDF.js library.

Reply:
I'm not sure I follow the logic here. The getRgb(..) call is the same as the original code and the only difference is how its stacked in the args array. Is there logic in the original getRgb() that asserts that it always is a 3 argument array coming out?

Perhaps a solution with rgb first and cmyk as the following args solves this
```js
var rgb = ColorSpace.singletons.cmyk.getRgb(args, 0);
args = [rgb[0], rgb[1], rgb[2], args[0], args[1], args[2], args[3]]
```

Extension for handling the undefined case (if not handled anywhere else could be)
```js
args = [ 
             rgb[0] || 0,
             rgb[1] || 0, 
             rgb[2] || 0, 
             args[0] || 0, 
             args[1] || 0, 
             args[2] || 0, 
             args[3] || 0
]
```

Reply:
The "original" `args` array could contain *anything*, i.e. even un-clonable data, and you're now passing it as-is without any validation. With the current code, any unexpected data-structure would lead to an Error that we're able to handle gracefully within `PartialEvaluator.getOperatorList`, however with these changes it may instead cause a DataCloneError during `postMessage`.

Reply:
Will it not throw the error when parsing the invalid args in the getRgb(..)?

Reply:
Maybe, but given just how badly some PDF generators can corrupt a document you quite simply cannot depend on that. Sending non-validated data, in any form, is really a non-starter here (and elsewhere).